swiftsimio/accelerated.py

```diff
@@ -168,3 +180,148 @@ def index_dataset(handle: Dataset, mask_array: np.array) -> np.array:
     ranges = ranges_from_array(mask_array)
 
     return read_ranges_from_file(handle, ranges, output_size, output_type)
+
+
+################################ ALEXEI: playing around with better read_ranges_from_file implementation #################################
+
+def get_chunk_ranges(ranges, chunk_size, array_length):
+    chunk_ranges = []
+    for bound in ranges:
+        lower = int(np.floor(bound[0]/chunk_size))*chunk_size
```

Comment:
Please make sure that you format with black every time before you make a commit.

You should use `math` instead of `numpy` for single values - these will get compiled better by numba and should be faster than making calls out to numpy.

Aren't bound[0] and chunk_size both ints? In that case we can use integer division:

`lower = (bound[0] // chunk_size) * chunk_size`